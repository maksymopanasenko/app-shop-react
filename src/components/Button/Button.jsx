import './Button.scss';
import PropTypes from 'prop-types';

function Button({ text, disabled, onClick }) {
    return (
        <button className='btn' disabled={disabled} onClick={onClick} >
            {text}
        </button>
    );
}

Button.propTypes = {
    text: PropTypes.string.isRequired,
    onClick: PropTypes.func
}

export default Button;