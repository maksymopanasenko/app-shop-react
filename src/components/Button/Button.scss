.btn {
    padding: 10px 20px;
    border: none;
    border-radius: 5px;
    color: #fff;
    background: #564356;
    transition: .3s;

    &:hover {
        background: #3d2f3d;
    }

    &:disabled {
        background: #4c474c;
        color: #aaaaaa;
    }
}