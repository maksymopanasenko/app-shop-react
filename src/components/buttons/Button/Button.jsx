import React from 'react';
import './Button.scss';
import PropTypes from 'prop-types';

function Button({ text, disabled, onClick }) {
    return (
        <button className='btn' disabled={disabled} onClick={onClick} data-testid="Button">
            {text}
        </button>
    );
}

Button.propTypes = {
    text: PropTypes.string.isRequired,
    onClick: PropTypes.func,
    disabled: PropTypes.bool
}

export default Button;