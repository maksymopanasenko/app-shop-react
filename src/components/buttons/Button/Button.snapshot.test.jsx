import React from "react";
import renderer from 'react-test-renderer';
import Button from "./Button";
import { expect, test } from "vitest";

test('Button snapshot', () => {
    const component = renderer.create(<Button text="Some text"/>);
    
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});