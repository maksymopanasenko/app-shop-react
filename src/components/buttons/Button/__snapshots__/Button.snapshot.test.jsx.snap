// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Button snapshot 1`] = `
<button
  className="btn"
  data-testid="Button"
>
  Some text
</button>
`;
